src/repl.zig

```diff
@@ -583,6 +622,122 @@ pub fn ReplType(comptime MessageBus: type) type {
+
+                        // Move cursor to the start of the current row, clear the screen from start
+                        // to end. Then print the leading buffer, currently selected completion and
+                        // trailing buffer(if any).
+                        terminal_screen.update_cursor_position(
+                            -@as(isize, @intCast(buffer_index)),
+                        );
+                        const buffer_row = terminal_screen.cursor_row;
+
+                        terminal_screen.update_cursor_position(
+                            @as(isize, @intCast(match_prefix.items.len + current_match.len)),
+                        );
+
+                        // \x1b[n;mH - Moves the cursor to row n, column m.
+                        // \x1b[J - Clear the screen from current cursor position to end.
+                        try repl.terminal.print("\x1b[{};1H\x1b[J{s}{s}{s}{s}\x20\x08", .{
+                            buffer_row,
+                            prompt,
+                            match_prefix.items,
+                            current_match,
+                            match_suffix.items,
+                        });
+
+                        // When cursor reaches the last row of terminal, scroll content upward
+                        // to create space for displaying completion menu.
+                        // \x1b[nS - Scroll whole page up by n lines; defualt 1 line.
+                        // \x1b[nA - Moves cursor up n cells; default 1 cell up.
+                        if (terminal_screen.cursor_row == terminal_screen.rows) {
+                            try repl.terminal.print("\x1b[S\x1b[A", .{});
+                        }
+
+                        // Position cursor for completion menu.
+                        try repl.terminal.print("\x1b[{};1H", .{buffer_row + 1});
+
+                        // Display completion menu in the next line and highlight the currently
+                        // selected completion.
+                        var menu_width: usize = 0;
+                        for (matches.items) |match| {
+                            menu_width += match.len + 2;
+
+                            // \x1b[7m - Highlights the text by inverting the color. Inverts
+                            //           the text and background color.
+                            // \x1b[0m - Reset the style property of text. In this case, resets
+                            //           the effect of \x1b[7m - color inversion.
+                            if (std.mem.eql(u8, match, current_match)) {
+                                try repl.terminal.print("\x20\x1b[7m{s}\x1b[0m\x20", .{match});
+                            } else {
+                                try repl.terminal.print("\x20{s}\x20", .{match});
+                            }
+                        }
+
+                        // If the completion menu overflow past the last row of terminal screen,
```

Comment:
`overflow` -> `overflows`

Reply:
this comment is fixed. if you see anything else, please let me know.